usaspending_api/api_contracts/contracts/v2/agency/toptier_code/obligations_by_type.md

```diff
@@ -0,0 +1,58 @@
+FORMAT: 1A
+HOST: https://api.usaspending.gov
+
+# Federal Obligations [/api/v2/agency/{toptier_code}/obligations_by_type/{?fiscal_year}]
+
+This endpoint is used to power USAspending.gov's agency profile pages.
+
+## GET
+
+This endpoint returns a breakdown of obligations by award type (contracts, IDVs, grants, loans, direct payments, other) within a fiscal year.
+
++ Parameters
+    + `toptier_code`: 086 (required, number)
+        The toptier code of an agency (could be a CGAC or FREC) so only numeric character strings of length 3-4 are accepted.
+    + `fiscal_year`: 2017 (required, number)
+        The fiscal year that you are querying data for.
+        
++ Response 200 (application/json)
+    + Attributes (object)
+        + `award_obligations` (required, number)
+        + `results` (required, array[ObligationSubtotals], fixed-type)
+
+    + Body
+
+            {
+                "award_obligations": 39999999.96,
+                "results": [
+                    {
+                        "type": "contracts",
+                        "obligated_amount": 9999999.99
+                    },
+                    {
+                        "type": "idvs",
+                        "obligated_amount": 9999999.99
+                    },
+                    {
+                        "type": "direct_payments",
+                        "obligated_amount": 9999999.99
+                    },
+                    {
+                        "type": "grants",
+                        "obligated_amount": 9999999.99
+                    }
+                ]
+            }
+
+# Data Structures
+
+## ObligationSubtotals (object)
++ `type` (required, enum[string])
+    + Members
+        + `contracts`
+        + `idvs`
+        + `grants`
+        + `loans`
+        + `direct_payments`
+        + `other`
++ `obligated_amount` (required, number)
```

Comment:
can these values be null?

Reply:
Lizzie & I discussed; our understanding was neither null nor 0 would be included in the results. But can they be negative?

Reply:
I don't see a reason why these values cannot be negative or zero. We don't have any syntax in API Blueprint to limit numeric values to ranges (outside of enums, which are of course limited)

Reply:
Yeah my question was if an agency has no spending for a category, will we:
A) excluded that category from `results` (in this case, they should not be nullable)
B) always include all categories, but with an obligated amount of zero or null 
FE can work with it either way. We'll also handle negative values if they exist (up to design what that will look like...)

Reply:
I assumed all award categories would always be returned in the response like other endpoints. But the fact that it is an array does change things since any category without spending could be dropped instead of providing $0. If that is what Brett wants and Lizzie is OK then we're on the same page.

@brett-varney did you think about providing the results in a sorted order? I recommend it and having it documented in the comment alongside the note on $0 results

Reply:
@tony-sappe Makes sense, thanks